nginx-controller/nginx/configurator.go

```diff
@@ -190,6 +192,17 @@ func (cnf *Configurator) createConfig(ingEx *IngressEx) Config {
 	if proxyReadTimeout, exists := ingEx.Ingress.Annotations["nginx.org/proxy-read-timeout"]; exists {
 		ingCfg.ProxyReadTimeout = proxyReadTimeout
 	}
+	if proxyHideHeaders, exists, err := GetMapKeyAsStringSlice(ingEx.Ingress.Annotations, "nginx.org/proxy-hide-headers", ingEx.Ingress); exists {
+		if err != nil {
+			glog.Error(err)
+		} else {
+			if ingCfg.ProxyHideHeaders == nil || len(ingCfg.ProxyHideHeaders) == 0 {
+				ingCfg.ProxyHideHeaders = proxyHideHeaders
+			} else {
+				ingCfg.ProxyHideHeaders = append(ingCfg.ProxyHideHeaders, proxyHideHeaders...)
```

Comment:
If the `nginx.org/proxy-hide-headers` annotations exists, it's better to overwrite the value provided by the configmap. Such a behavior is consistent with the behaviour of the `	proxy_hide_header` NGINX directive.